From 01964795bd433bee5266862936827e953df090d6 Mon Sep 17 00:00:00 2001
From: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Date: Wed, 20 Aug 2025 01:05:59 +0300
Subject: [PATCH 8/8] kgsl: fix building with 6.12 +
 qcom_scm_set_gpu_smmu_aperture()

Fix building with upstream kernel with qcom_scm_set_gpu_smmu_aperture()
function backported to it by checking for the CONFIG_QCOM_SCM_ADDON.

Signed-off-by: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Upstream-Status: Submitted [https://github.com/qualcomm-linux/kgsl/pull/9]
---
 kgsl_iommu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kgsl_iommu.c b/kgsl_iommu.c
index bd415fa2c050..837241ce3409 100644
--- a/kgsl_iommu.c
+++ b/kgsl_iommu.c
@@ -1334,7 +1334,7 @@ static void _enable_gpuhtw_llc(struct kgsl_mmu *mmu, struct iommu_domain *domain
 		iommu_set_pgtable_quirks(domain, IO_PGTABLE_QUIRK_ARM_OUTER_WBWA);
 }
 
-#if (KERNEL_VERSION(6, 13, 0) <= LINUX_VERSION_CODE)
+#if (KERNEL_VERSION(6, 13, 0) <= LINUX_VERSION_CODE) || !defined (CONFIG_QCOM_SCM_ADDON)
 int _kgsl_set_smmu_aperture(u32 num_context_bank)
 {
 	return qcom_scm_set_gpu_smmu_aperture(num_context_bank);
-- 
2.47.2

